include/onnxruntime/core/graph/graph.h

```diff
@@ -734,8 +734,10 @@ class Graph {
   Any graph inputs not in this list will be appended to the ordered graph input list, in the order that they were first
   used by Nodes (i.e. the order of Node creation implicitly determines the ordering).
   @remarks If the Graph was loaded from a GraphProto this has no effect.*/
-  void SetInputOrder(const std::vector<const NodeArg*> inputs) {
-    graph_input_order_ = inputs;
+  void SetInputs(const std::vector<const NodeArg*> inputs) {
+    graph_inputs_including_initializers_ = inputs;
+    graph_inputs_manually_set = true;
```

Comment:
Shouldn't this be graph_inputs_excluding_initializers_ as this is the list of explicit inputs?

Reply:
Actually even that probably isn't enough. This list could have optional inputs in it so doesn't map 1:1 with graph_inputs_including_initializers_ or graph_inputs_excluding_initializers_.

As such I don't think you can use it directly. 

---
In reply to: [278757193](https://github.com/Microsoft/onnxruntime/pull/905#discussion_r278757193) [](ancestors = 278757193)

Reply:
no, graph_inputs_including_initializers should be the one mapping to graphproto.inputs.